tests/system/small/test_index.py

```diff
@@ -320,6 +321,30 @@ def test_index_to_series(
     pd.testing.assert_series_equal(bf_result, pd_result)
 
 
+@pytest.mark.parametrize("index_arg", [True, False])
+@pytest.mark.parametrize("name_arg", [None, "food"])
+def test_index_to_frame(index_arg, name_arg):
+    pd_idx: pd.Index = pd.Index(
+        ["Ant", "Bear", "Cow"], name="animal", dtype="string[pyarrow]"
+    )
+    bf_idx = indexes.Index(["Ant", "Bear", "Cow"], name="animal")
+
+    if name_arg is None:
```

Comment:
It doesn't need the if else condition. Just

```
pd_df = pd_idx.to_frame(index=index_arg, name=name_arg)
bf_df = bf_idx.to_frame(index=index_arg, name=name_arg)
```

is good enough

Reply:
Pandas implementation of `name` arg handling is slightly different. If we set `name=None` for pandas, it will create a DataFrame with column names as string `None`, since technically the default is `lib.nodefault` (code pointer: https://github.com/pandas-dev/pandas/blob/v2.2.2/pandas/_libs/lib.pyi#L32), whereas use `None`: https://github.com/pandas-dev/pandas/blob/v2.2.2/pandas/core/indexes/base.py#L1607-L1666

Reply:
Happy to add something similar to BigFrames if you think it is better to exactly mirror pandas functionality

Reply:
Ah, that's a weird behavior of pandas. Thanks for pointing out. Then we shouldn't let the default to be None. But should have sth similar. @TrevorBergeron